packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -64,22 +64,45 @@ export interface ITree extends IChannel {
 	): TreeView<TRoot>;
 }
 
+/**
+ * Options when schematizing a tree.
+ * @public
+ */
+export interface ITreeConfigurationOptions {
+	/**
+	 * If `true`, the tree will validate new content against its stored schema at insertion time
+	 * and throw an error if the new content doesn't match the expected schema.
+	 */
+	enableSchemaValidation: boolean;
+}
+
+const defaultTreeConfigurationOptions: ITreeConfigurationOptions = {
+	enableSchemaValidation: false,
+};
+
 /**
  * Configuration for how to {@link ITree.schematize|schematize} a tree.
  * @public
  */
 export class TreeConfiguration<TSchema extends ImplicitFieldSchema = ImplicitFieldSchema> {
+	public readonly options: ITreeConfigurationOptions;
+
 	/**
 	 * @param schema - The schema which the application wants to view the tree with.
 	 * @param initialTree - A function that returns the default tree content to initialize the tree with iff the tree is uninitialized
 	 * (meaning it does not even have any schema set at all).
 	 * If `initialTree` returns any actual node instances, they should be recreated each time `initialTree` runs.
 	 * This is because if the config is used a second time any nodes that were not recreated could error since nodes cannot be inserted into the tree multiple times.
+	 * @param enableSchemaValidation - If true, the tree will validate new content against its stored schema at insertion time
+	 * and throw an error if the new content doesn't match the expected schema.
 	 */
 	public constructor(
 		public readonly schema: TSchema,
 		public readonly initialTree: () => InsertableTreeFieldFromImplicitField<TSchema>,
-	) {}
+		options?: ITreeConfigurationOptions,
```

Comment:
Since we have a default that we fall back to, would it make sense to take this as a `Partial<ITreeConfigurationOptions>`? Alternatively, it might make sense for `enableSchemaValidation` to be optional in the interface, and document its default value, rather than having a default config object.

Reply:
Yeah, good call. Updating.